File: android/app/src/main/java/com/wikiart/model/ArtistCategory.kt
package com.wikiart.model

import com.wikiart.R
import com.wikiart.CategoryItem

/**
 * Categories for browsing artists.
 * Mirrors the iOS ArtistCategory enum.
 */
enum class ArtistCategory(val path: String, private val hasSections: Boolean) : CategoryItem {
    ALPHABETICAL("Alphabet", false),
    ART_MOVEMENT("artists-by-Art-Movement", true),
    SCHOOL("artists-by-painting-school", true),
    GENRE("artists-by-genre", true),
    FIELD("artists-by-field", true),
    NATION("artists-by-nation", true),
    CENTURIES("artists-by-century", true),
    CHRONOLOGY("chronological-artists", false),
    POPULAR("popular-artists", false),
    FEMALE("female-artists", false),
    RECENT("recently-added-artists", false),
    INSTITUTIONS("artists-by-art-institution", true);

    fun hasSections(): Boolean = hasSections

    override fun toString(): String = when (this) {
        ALPHABETICAL -> "Alphabet"
        ART_MOVEMENT -> "Art Movement"
        SCHOOL -> "Painting School"
        GENRE -> "Genre"
        FIELD -> "Field"
        NATION -> "Nation"
        CENTURIES -> "Century"
        CHRONOLOGY -> "Chronological"
        POPULAR -> "Popular"
        FEMALE -> "Female"
        RECENT -> "Recent"
        INSTITUTIONS -> "Art Institution"
    }

    fun nameRes(): Int = when (this) {
        ALPHABETICAL -> R.string.artist_category_alphabet
        ART_MOVEMENT -> R.string.artist_category_art_movement
        SCHOOL -> R.string.artist_category_school
        GENRE -> R.string.artist_category_genre
        FIELD -> R.string.artist_category_field
        NATION -> R.string.artist_category_nation
        CENTURIES -> R.string.artist_category_century
        CHRONOLOGY -> R.string.artist_category_chronology
        POPULAR -> R.string.artist_category_popular
        FEMALE -> R.string.artist_category_female
        RECENT -> R.string.artist_category_recent
        INSTITUTIONS -> R.string.artist_category_institution
    }

    fun iconRes(): Int = R.drawable.ic_artists
}

File: WikiArt3.0/Models/ArtistCategory.swift
//
//  ArtistCategories.swift
//  WikiArt3.0
//
//  Created by Waleed Azhar on 2019-10-22.
//  Copyright © 2019 Waleed Azhar. All rights reserved.
//

import Foundation

enum ArtistCategory: String {
    static let all: [ArtistCategory] =  [ .popular, .female, .recent, .artMovement, .school, .genre, .field, .nation, .centuries, .chronology, .institutions]
    static let categoriesWith: [ArtistCategory] = [.artMovement, .school, .genre, .field, .nation, .centuries, .institutions]
    
    case alphabetical = "Alphabet"
    case artMovement = "artists-by-Art-Movement"
    case school = "artists-by-painting-school"
    case genre = "artists-by-genre"
    case field = "artists-by-field"
    case nation = "artists-by-nation"
    case centuries = "artists-by-century"
    case chronology = "chronological-artists"
    case popular = "popular-artists"
    case female = "female-artists"
    case recent = "recently-added-artists"
    case institutions = "artists-by-art-institution"
}

extension ArtistCategory {

    var title: String {
        switch self {
        case .alphabetical:
            return NSLocalizedString("Alphabet", comment: "")
        case .artMovement:
            return NSLocalizedString("Art Movement", comment: "")
        case .school:
            return NSLocalizedString("Painting School", comment: "")
        case .genre:
            return NSLocalizedString("Genre", comment: "")
        case .field:
            return NSLocalizedString("Field", comment: "")
        case .nation:
            return NSLocalizedString("Nation", comment: "")
        case .centuries:
            return NSLocalizedString("Century", comment: "")
        case .chronology:
            return NSLocalizedString("Chronological", comment: "")
        case .popular:
            return NSLocalizedString("Popular", comment: "")
        case .female:
            return NSLocalizedString("Female", comment: "")
        case .recent:
            return NSLocalizedString("Recent", comment: "")
        case .institutions:
            return NSLocalizedString("Art Institution", comment: "")
        }
    }
    
    var hasSections: Bool {
        switch self {
        case .alphabetical:
            return false
        case .artMovement:
            return true
        case .school:
            return true
        case .genre:
            return true
        case .field:
            return true
        case .nation:
            return true
        case .centuries:
            return true
        case .chronology:
            return false
        case .popular:
            return false
        case .female:
            return false
        case .recent:
            return false
        case .institutions:
            return true
        }
    }
    
    var icon: Icon {
        switch self {
        case .popular:
            return .popular
        case .genre:
            return .genre
        case .alphabetical:
            return .blank
        case .artMovement:
            return .style
        case .school:
            return .school
        case .field:
            return .field
        case .nation:
            return .nation
        case .centuries:
            return .century
        case .chronology:
            return .chronological
        case .female:
            return .female
        case .recent:
            return .recent
        case .institutions:
            return .institution
        }
    }
   
}


